src/js/components/stock-movement-wizard/inbound/AddItemsPage.jsx

```diff
@@ -619,24 +635,53 @@ class AddItemsPage extends Component {
     this.props.showSpinner();
 
     this.saveRequisitionItemsInCurrentStep(lineItems)
-      .then((resp) => {
+      .then(async (resp) => {
         let values = formValues;
         if (resp) {
           values = { ...formValues, lineItems: resp.data.data.lineItems };
         }
 
-        if (_.some(values.lineItems, (item) => item.inventoryItem
-          && item.expirationDate !== item.inventoryItem.expirationDate)) {
-          if (_.some(values.lineItems, (item) => item.inventoryItem.quantity && item.inventoryItem.quantity !== '0')) {
+        const hasExpiryMismatch = values.lineItems.some(
+          (item) => item.inventoryItem && item.expirationDate !== item.inventoryItem.expirationDate,
+        );
+
+        if (hasExpiryMismatch) {
+          const hasNonZeroQuantity = values.lineItems.some(
+            (item) => item.inventoryItem?.quantity && item.inventoryItem.quantity !== '0',
+          );
+
+          if (hasNonZeroQuantity) {
             this.props.hideSpinner();
-            this.confirmInventoryItemExpirationDateUpdate(() =>
-              this.updateInventoryItemsAndTransitionToNextStep(values, lineItems));
-          } else {
-            this.updateInventoryItemsAndTransitionToNextStep(values, lineItems);
+
+            const itemsWithMismatchedExpiry = values.lineItems
+              .filter(
+                (item) =>
+                  item.inventoryItem &&
+                  item.inventoryItem.expirationDate !== item.expirationDate &&
+                  item.inventoryItem.quantity &&
+                  item.inventoryItem.quantity !== '0',
+              )
+              .map((item) => ({
+                code: item.product?.productCode,
+                product: item.product,
+                lotNumber: item.lotNumber,
+                previousExpiry: item.inventoryItem.expirationDate,
+                newExpiry: item.expirationDate,
+              }));
+
+            if (itemsWithMismatchedExpiry.length > 0) {
+              const shouldUpdateExpirationDate = await
+              this.confirmExpirationDateSave(itemsWithMismatchedExpiry);
```

Comment:
the formatting looks weird here. don't break the await calls to another line. If you need to move it due to the eslint error as the line is too long, move it together with the `await`.